### PR TITLE
Refactor template cmds packages to each sub-cmd

### PR DESCRIPTION
**Pull request by chloechien**

### CODE REVIEW

File: templates/commands/render/render_action.go [hunk 1]
```diff
@@ -146,19 +147,19 @@ func templateFuncs() template.FuncMap {
...
 		"trimPrefix":        strings.TrimPrefix,
 		"trimSuffix":        strings.TrimSuffix,
 		"trimSpace":         strings.TrimSpace,
-		"toSnakeCase":       toSnakeCase,
-		"toLowerSnakeCase":  toLowerSnakeCase,
-		"toUpperSnakeCase":  toUpperSnakeCase,
-		"toHyphenCase":      toHyphenCase,
-		"toLowerHyphenCase": toLowerHyphenCase,
-		"toUpperHyphenCase": toUpperHyphenCase,
+		"toSnakeCase":       utils.ToSnakeCase,
```
Comment (drevell) [inline templates/commands/render/render_action.go:157]: Could we leave these in the "render" package? Is there an important reason to move them into utils?

File: templates/commands/render/render_action_foreach.go [hunk 2]
```diff
@@ -34,7 +35,7 @@ func actionForEach(ctx context.Context, fe *model.ForEach, sp *stepParams) error
 			return err
 		}
 	} else {
-		if err := celCompileAndEval(ctx, sp.scope, *fe.Iterator.ValuesFrom, &values); err != nil {
+		if err := utils.CelCompileAndEval(ctx, sp.scope, *fe.Iterator.ValuesFrom, &values); err != nil {
 			return err
```
Comment (drevell) [inline templates/commands/render/render_action_foreach.go:39]: Regarding the lint error that's triggered on this line:

```
Check failure on line 39 in templates/commands/render/render_action_foreach.go

GitHub
Actions
/ go_lint / golangci

error returned from external package is unwrapped: sig: func github.com/abcxyz/abc/templates/utils.CelCompileAndEval(ctx context.Context, scope *github.com/abcxyz/abc/templates/utils.Scope, expr github.com/abcxyz/abc/templates/model.valWithPos[string], outPtr any) error (wrapcheck)
```

You can add `//nolint:wrapcheck` at the end of the line to silence it. This is appropriate beacuse CelCompileAndEval returns an error that doesn't need to be wrapped.

File: templates/utils/template_funcs.go [hunk 3]
```diff
@@ -12,7 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package commands
+// Package utils contains the utility function for template commands.
+
+package utils
```
Comment (drevell) [inline templates/utils/template_funcs.go:17]: Can we name this "common" rather than "utils"? Seems a bit more descriptive

**Review by drevell — CHANGES_REQUESTED** [review]

File: templates/commands/render/render_action.go (hunk 1 above)
Comment (drevell) [reply templates/commands/render/render_action.go:157]: Never mind, I understand now.

File: templates/commands/render/render_action_foreach.go (hunk 2 above)
Comment (chloechien) [reply templates/commands/render/render_action_foreach.go:39]: Thank you!

File: templates/utils/template_funcs.go (hunk 3 above)
Comment (chloechien) [reply templates/utils/template_funcs.go:17]: Done.

File: cmd/abc/abc.go [hunk 4]
```diff
@@ -38,18 +39,18 @@ var rootCmd = func() *cli.RootCommand {
...
+							return &render.RenderCommand{}
 						},
 						"test": func() cli.Command {
 							return &cli.RootCommand{
 								Name:        "test",
 								Description: "subcommands for validating template rendering with golden tests",
 								Commands: map[string]cli.CommandFactory{
 									"record": func() cli.Command {
-										return &commands.TestRecordCommand{}
+										return &test.TestRecordCommand{}
```
[the remark this answers is not in the document]
Comment (chloechien) [reply cmd/abc/abc.go]: Good to know! Updated.

**Review by drevell — APPROVED** [review]

**Pull request merged**